src/Tools/AI Test Toolkit/src/Logs/AITRunHistory.Page.al

```diff
@@ -0,0 +1,222 @@
+
+page 149036 "AIT Run History"
+{
+    Caption = 'AI Test Suite Run History';
+    PageType = Worksheet;
+    ApplicationArea = All;
+    UsageCategory = None;
+    SourceTable = "AIT Run History";
+    SourceTableTemporary = true;
+    InsertAllowed = false;
+    ModifyAllowed = false;
+    DeleteAllowed = false;
+
+    layout
+    {
+        area(Content)
+        {
+            group(General)
+            {
+                Caption = 'General';
+
+                field("Test Suite Code"; TestSuiteCode)
+                {
+                    Caption = 'Test Suite Code';
+                    ToolTip = 'Specifies the code of the test suite to view results for.';
+                    TableRelation = "AIT Test Suite".Code;
+
+                    trigger OnValidate()
+                    begin
+                        UpdateRunHistory();
+                    end;
+                }
+
+                field("View By"; ViewBy)
+                {
+                    Caption = 'View By';
+                    ToolTip = 'Specifies whether to view results by tag or version.';
+
+                    trigger OnValidate()
+                    begin
+                        UpdateRunHistory();
+                    end;
+                }
+
+                field("Apply Filter"; ApplyLineFilter)
+                {
+                    Caption = 'Apply Filter';
+                    ToolTip = 'Specifies whether to filter results to a specific line or see results of the suite as a whole.';
+
+                    trigger OnValidate()
+                    begin
+                        UpdateRunHistory();
+                    end;
+                }
+            }
+
+            group("Line Filter")
+            {
+                Caption = 'Test Line Filter';
+                Visible = ApplyLineFilter;
+
+                field("Line No."; LineNoFilter)
+                {
+                    Caption = 'Test Line';
+                    ToolTip = 'Specifies the line to filter to.';
+                    Editable = false;
+
+                    trigger OnAssistEdit()
+                    begin
+                        LookupTestMethodLine();
+                    end;
+                }
+            }
+
+            group(History)
+            {
+                repeater("Run History")
+                {
+                    field(Version; Rec.Version)
+                    {
+                        Visible = ViewBy = ViewBy::Version;
+                        Caption = 'Version';
+                        ToolTip = 'Specifies the version.';
+                    }
+                    field(Tag; Rec.Tag)
+                    {
+                        Caption = 'Tag';
+                        ToolTip = 'Specifies the tag of the version.';
+                    }
+                    field("No. of Tests - By Version"; Rec."No. of Tests Executed")
+                    {
+                        Visible = ViewBy = ViewBy::Version;
```

Comment:
Instead of trying to do this ViewBy stuff, why not just use filters normally? 
Then you dont need the table to contain both things either? And people can use BC platform filters as normal.

Reply:
The number of records that we need to display is different based on the view. If it's version, then we need to get the aggregated numbers for each version, but you can have multiple versions with the same tag. 

So, resolving this as it's not possible to achieve the desired functionality with just normal filters.